ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/project/type/ProjectTypeImpl.java

```diff
@@ -105,9 +115,29 @@ public boolean isTypeOf(String typeId) {
      * @return attribute by name
      */
     public Attribute getAttribute(String name) {
-        return attributes.get(name);
+        for(Attribute attr : attributes) {
+            if(attr.getName().equals(name))
+                return attr;
+        }
+        return null;
     }
 
+    @Override
+    public int hashCode() {
+        return id.hashCode();
+    }
+
+    @Override
+    public boolean equals(Object obj) {
```

Comment:
*if*-blocks should be formatted in accordance with codestyle: https://github.com/codenvy/codenvy-dev-resources/wiki/Formatting#non-empty-blocks